sea-orm-macros/src/derives/column.rs

```diff
@@ -28,36 +28,24 @@ pub fn impl_default_as_str(ident: &Ident, data: &Data) -> syn::Result<TokenStrea
         .map(|v| {
             let mut column_name = v.ident.to_string().to_snake_case();
             for attr in v.attrs.iter() {
-                if let Some(ident) = attr.path.get_ident() {
-                    if ident != "sea_orm" {
-                        continue;
-                    }
-                } else {
+                if !attr.path().is_ident("sea_orm") {
                     continue;
                 }
-                if let Ok(list) = attr.parse_args_with(Punctuated::<Meta, Comma>::parse_terminated)
-                {
-                    for meta in list.iter() {
-                        if let Meta::NameValue(nv) = meta {
-                            if let Some(name) = nv.path.get_ident() {
-                                if name == "column_name" {
-                                    if let Lit::Str(litstr) = &nv.lit {
-                                        column_name = litstr.value();
-                                    }
-                                }
-                                if name == "table_name" {
-                                    if let Lit::Str(litstr) = &nv.lit {
-                                        column_name = litstr.value();
-                                    }
-                                }
-                            }
-                        }
+                attr.parse_nested_meta(|meta| {
+                    if meta.path.is_ident("column_name") || meta.path.is_ident("table_name") {
+                        column_name = meta.value()?.parse::<LitStr>()?.value();
+                    } else {
+                        // Reads the value expression to advance the parse stream.
+                        // Some parameters, such as `primary_key`, do not have any value,
+                        // so ignoring an error occurred here.
+                        let _: Option<Expr> = meta.value().and_then(|v| v.parse()).ok();
```

Comment:
For some reason, we both `#[sea_orm(column_name = "col_name")]` and `#[sea_orm(table_name = "col_name")]` override the `str` value of column iden.

We should drop the use of `#[sea_orm(table_name = "col_name")]`

https://github.com/SeaQL/sea-orm/pull/1713/commits/1fba661468db20dbf228d74cbda158d73e7fe721